### PR TITLE
feat: allowance action edit

### PR DESCRIPTION
**Pull request by varshamenon4**
[COSMO-394](https://2u-internal.atlassian.net/browse/COSMO-394)

### CODE REVIEW

File: src/pages/ExamsPage/components/AllowanceListActions.jsx [hunk 1]
```diff
@@ -10,8 +10,53 @@ import {
...
+import { useDeleteAllowance, useEditAllowance } from '../hooks';
+
+// todo: add onEdit
+const EditModal = ({
+  isOpen, onCancel, onEdit,
+}) => {
+  const [ additionalTimeError, setAdditionalTimeError ] = useState(false);
+  const { formatMessage } = useIntl();
+
+  return (
```
Comment (varshamenon4) [inline src/pages/ExamsPage/components/AllowanceListActions.jsx:25]: This part is still very much in progress. Currently trying to figure out why I can't open the modal. I'm wondering if it has something to do with the way the props are passed in? I'm referencing the AddAllowanceModal, but I think I need to set some sort of initial state maybe?

File: src/pages/ExamsPage/components/AllowanceListActions.jsx [hunk 2]
```diff
@@ -10,8 +10,53 @@ import {
...
+import { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance } from '../hooks';
+
+// todo: add onEdit
+const EditModal = ({
+  isOpen, onCancel, onEdit,
+}) => {
+  const [ additionalTimeError, setAdditionalTimeError ] = useState(false);
```
Comment (varshamenon4) [inline src/pages/ExamsPage/components/AllowanceListActions.jsx:22]: Not in use yet, but adding to handle errors.

Comment (varshamenon4) [inline src/pages/ExamsPage/components/AllowanceListActions.jsx:22]: Should the Additional time only take in time in minutes or should it also handle time multiplier? Wasn't sure based on the figma here: https://www.figma.com/design/PXtvq1xV4R7dkVlLuqCE8s/Allowances?node-id=0-1&t=kWnGHiyO6EYb5OLP-0

**Review by varshamenon4 — COMMENTED** [review]
Still WIP, but some questions.

File: src/pages/ExamsPage/components/AllowanceListActions.jsx (hunk 2 above)
Comment (Zacharis278) [reply src/pages/ExamsPage/components/AllowanceListActions.jsx:22]: the multiplier isn't persisted so I think the only way to edit is change the raw value. The multiplier is mostly needed for doing multiple allowances at once.

File: src/pages/ExamsPage/components/AllowanceListActions.jsx (hunk 1 above)
Comment (Zacharis278) [reply src/pages/ExamsPage/components/AllowanceListActions.jsx:25]: A few options to debug. If you just set isOpen to true in the render does it work? That would rule layout/render issues. What value is this component getting passed in for isOpen?

File: src/pages/ExamsPage/data/api.js [hunk 3]
```diff
@@ -50,3 +50,10 @@ export async function createAllowance(courseId, data) {
   const response = await getAuthenticatedHttpClient().post(url, data);
   return response.data;
 }
+
+export async function editAllowance(courseId, allowanceId, extraTimeMins) {
```
Comment (Zacharis278) [inline src/pages/ExamsPage/data/api.js:54]: this is just the same code as create, maybe we just call create in the edit hook? Kind of saves us from adding and testing duplicate code.
